girder-dandi-archive/tests/rest/test_dandiset_ownership.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+import pytest
+
+from pytest_girder.assertions import assertStatusOk
+
+pytestmark = pytest.mark.plugin("dandi_archive")
+
+
+@pytest.fixture
+def admin_created_dandiset(server, admin):
+    resp = server.request(
+        path="/dandi",
+        method="POST",
+        user=admin,
+        params={
+            "name": "admin_created_dandiset",
+            "description": "Admin created this test dandiset.",
+        },
+    )
+    assertStatusOk(resp)
+    return resp.json
+
+
+@pytest.fixture
+def multi_owner_dandiset(server, admin, user, admin_created_dandiset):
```

Comment:
It's not an issue as this currently stands, but using the `multi_owner_dandiset` fixture will corrupt the `admin_created_dandiset` fixture. I can't picture a single test that would need both, but it feels like a trap. Considering this fixture is only used in one test, it might be better to remove this fixture and just add the permissions in the test? I don't feel very strongly.

Reply:
The main reason I did that is because the remove test basically contained the entirety of the add test, and I didn't like that pattern. 

I was curious about this, as I had thought that fixtures were created new and independently for each test, so I tried testing it. What I did was replace the `admin_created_dandiset` argument in `test_initial_dandiset_owners` with `multi_owner_dandiset`, and then ran the tests. The tests run in the order of:

1. `test_initial_dandiset_owners`
2. `test_add_dandiset_owners`
3. `test_remove_dandiset_owners`

I didn't replace the fixture argument of `admin_created_dandiset` in `test_add_dandiset_owners`, so if it were to corrupt the `admin_created_dandiset` fixture, in that test it should show two users as being owners on the dandiset, instead of just 1. However, when I did this, only the 1 user showed.

Furthermore if I just printed the drafts collection in each test, I get different ID each time. I know this is a lot more than you asked for, but from looking around online and testing myself, it seems like mutation of the fixtures won't actually occur, so I'd be fine leaving it as is. If there's something here I'm missing please lmk.

Reply:
Sorry, I should have been more clear. The clobbering will occur when you have a test that requests both the `multi_owner_dandiset` fixture and the `admin_created_dandiset` at the same time, e.g.
```
def broken_test(admin_created_dandiset, multi_owner_dandiset):
  assert admin_created_dandiset["_id"] != multi_owner_dandiset["_id"] # fails
```
Both fixtures refer to the same dandiset, which now has multiple owners. The way it's written now will work perfectly for any reasonable tests that I can think of. I'm not 100% happy with the way this is written, but all the alternatives I can think of have flaws of similar magnitude. Up to you.

Reply:
Ahh I see what you mean. In that case I agree they should probably be independent. It will result in some duplicate code but I'm okay with that. I'll make that change.